internal/app/machined/pkg/controllers/cri/registry_test.go

```diff
@@ -0,0 +1,88 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+package cri_test
+
+import (
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/suite"
+
+	"github.com/siderolabs/talos/internal/app/machined/pkg/controllers/cri"
+	"github.com/siderolabs/talos/internal/app/machined/pkg/controllers/ctest"
+	"github.com/siderolabs/talos/pkg/machinery/config/container"
+	"github.com/siderolabs/talos/pkg/machinery/config/types/v1alpha1"
+	"github.com/siderolabs/talos/pkg/machinery/constants"
+	"github.com/siderolabs/talos/pkg/machinery/resources/config"
+	crires "github.com/siderolabs/talos/pkg/machinery/resources/cri"
+)
+
+type ConfigSuite struct {
+	ctest.DefaultSuite
+}
+
+func (suite *ConfigSuite) TestRegistry() {
+	cfg := config.NewMachineConfig(container.NewV1Alpha1(&v1alpha1.Config{
```

Comment:
there's a missing case for missing machine config

Reply:
Added `TestRegistryNoMachineConfig`.